src/utils/SeriesOptions.ts

```
@@ -9,64 +20,117 @@ module n3Charts.Utils {
+      options.visible = Options.getBoolean(options.visible);
+
+      return options;
     }
 
+    /**
+     * Return the toggeled visibility without modifying
+     * the visibility property itself
+     */
     getToggledVisibility() {
       return !this.visible;
     }
 
-    parseType(js:any): string[] {
-      if (!js) {
-        return ['line'];
-      }
+    /**
+     * Return an array of valid types
+     */
+    sanitizeType(types: string[]) {
+      return types.filter((type) => {
+        return SeriesOptions.isValidType(type);
+      });
+    }
 
-      if (typeof js === 'string') {
-        return [js];
+    /**
+     * Return a valid axis key
+     */
+    sanitizeAxis(axis: string) {
+      if (['y'].indexOf(axis) === -1) {
+        throw TypeError(axis + ' is not a valid series option for axis.');
       }
 
-      return js;
+      return axis;
     }
 
+    /**
+     * Returns true if the series has a type column.
+     * Series of type column need special treatment,
+     * because x values are usually offset
+     */
     isAColumn() {
-      return this.type.indexOf(Options.SERIES_TYPES.COLUMN) > -1;
+      return this.hasType(SeriesOptions.TYPE.COLUMN);
     }
 
     getMainType() {
-      if (this.type.length === 1) {
-        return this.type[0];
-      }
-
-      var types = Options.SERIES_TYPES;
+      // @Sebastien: do we really need a main type?
+      // I saw we are just using it for defining classes,
+      // but this could be perfectly done for every type
+      d3.values(SeriesOptions.TYPE).forEach((type) => {
+        if (this.hasType(type)) {
+          return type;
+        }
+      });
 
-      if (this.type.indexOf(types.AREA) !== -1) {
-        return types.AREA;
-      }
-
-      if (this.type.indexOf(types.LINE) !== -1) {
-        return types.LINE;
-      }
+      // We should always find the type with the lines
+      // above and never reach this return statement
+      return this.type[0];
```

Then let's throw an error !